python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py

```diff
@@ -680,255 +680,215 @@ def is_non_asset_node(dbt_resource_props: Mapping[str, Any]):
+                # necessary for backcompat
+                **({"group": resource_group_props} if resource_group_props else {}),
+            }
+        ),
+        tags=translator.get_tags(resource_props),
+        kinds={"dbt", manifest.get("metadata", {}).get("adapter_type", "dbt")},
+        partitions_def=translator.get_partitions_def(resource_props),
+    )
 
-        asset_deps[unique_id] = set()
-        for parent_unique_id in dbt_resource_props.get("depends_on", {}).get("nodes", []):
-            parent_node_info = dbt_nodes[parent_unique_id]
-            # for metrics or ephemeral dbt models, BFS to find valid parents
-            if is_non_asset_node(parent_node_info):
-                visited = set()
-                replaced_parent_ids = set()
-                # make a copy to avoid mutating the actual dictionary
-                queue = list(parent_node_info.get("depends_on", {}).get("nodes", []))
-                while queue:
-                    candidate_parent_id = queue.pop()
-                    if candidate_parent_id in visited:
-                        continue
-                    visited.add(candidate_parent_id)
-
-                    candidate_parent_info = dbt_nodes[candidate_parent_id]
-                    if is_non_asset_node(candidate_parent_info):
-                        queue.extend(candidate_parent_info.get("depends_on", {}).get("nodes", []))
-                    elif _valid_parent_node(candidate_parent_info):
-                        replaced_parent_ids.add(candidate_parent_id)
-
-                asset_deps[unique_id] |= replaced_parent_ids
-            # ignore nodes which are not assets / sources
-            elif _valid_parent_node(parent_node_info):
-                asset_deps[unique_id].add(parent_unique_id)
-
-    frozen_asset_deps = {
-        unique_id: frozenset(asset_deps[unique_id])
-        for unique_id in sorted(asset_deps)  # sort to stabilize job snapshots
-    }
+    # add integration-specific metadata to the spec
+    spec = spec.merge_attributes(
+        metadata={
+            DAGSTER_DBT_MANIFEST_METADATA_KEY: DbtManifestWrapper(manifest=manifest),
+            DAGSTER_DBT_TRANSLATOR_METADATA_KEY: translator,
+            DAGSTER_DBT_UNIQUE_ID_METADATA_KEY: resource_props["unique_id"],
+        }
+    )
+    if translator.settings.enable_code_references:
+        if not project:
+            raise DagsterInvalidDefinitionError(
+                "enable_code_references requires a DbtProject to be supplied"
+                " to the @dbt_assets decorator."
+            )
 
-    return frozen_asset_deps
+        spec = spec.replace_attributes(
+            metadata=_attach_sql_model_code_reference(
+                existing_metadata=spec.metadata,
+                dbt_resource_props=resource_props,
+                project=project,
+            )
+        )
+    return spec
 
 
-def build_dbt_multi_asset_args(
+def build_dbt_specs(
```

Comment:
not public yet, but I think that we tend to call these "load" instead of "build" now?

Reply:
there's a pre-existing public method called `build_dbt_asset_specs` (basically just strips out the check specs), so I went with that -- maybe we do another renaming pass at some point (agree load seems more common), just don't think this particular method will be made public because it has a tuple return type so feels ok to keep it as is for now

Reply:
seems reasonable; forgot that was public